include/Utility.hpp

```diff
@@ -39,20 +39,18 @@ template <typename T> std::pair<T, size_t> minimum(const std::vector<T> &vec)
 
 double convert(const std::string &str);
 
-template <typename T> double ClosestSamplingPoint(const std::vector<T> &vec, T const &val)
+inline double ClosestSamplingPoint(const std::vector<double> &vec, double const &val)
```

Comment:
what does inline do?

Reply:
it replaces the function call with the code itself, needed to either make it inline due to compiler errors of multiple definitions across files where that wasn't true.